bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -133,15 +133,116 @@ Host createHost(String name, CaseDbTransaction trans) throws TskCoreException {
 					throw new TskCoreException(String.format("Error adding host with name = %s and unable to rollback", name), ex);
 				}
 			}
-			
+
 			// It may be the case that the host already exists, so try to get it.
 			Optional<Host> optHost = getHost(name, connection);
 			if (optHost.isPresent()) {
 				return optHost.get();
 			}
 			throw new TskCoreException(String.format("Error adding host with name = %s", name), ex);
 		}
-	}	
+	}
+
+	/**
+	 * Updates host in database based on the host object provided.
+	 *
+	 * @param newHost The host to be updated.
+	 *
+	 * @return The newly returned host.
+	 *
+	 * @throws TskCoreException
+	 * @throws IllegalArgumentException
+	 */
+	public Host updateHost(Host newHost) throws TskCoreException, IllegalArgumentException {
+		if (newHost == null) {
+			throw new IllegalArgumentException("No host argument provided.");
+		} else if (newHost.getName() == null) {
+			throw new IllegalArgumentException(String.format("Host with id %d has no name", newHost.getId()));
+		}
+
+		CaseDbConnection connection = this.db.getConnection();
+		db.acquireSingleUserCaseWriteLock();
+		try {
+			String hostInsertSQL = "UPDATE tsk_hosts \n"
+					+ "SET name = ? \n"
+					+ "WHERE id = ?";
+
+			PreparedStatement preparedStatement = connection.getPreparedStatement(hostInsertSQL, Statement.RETURN_GENERATED_KEYS);
+
+			preparedStatement.clearParameters();
+			preparedStatement.setString(1, newHost.getName());
+			preparedStatement.setLong(2, newHost.getId());
+
+			connection.executeUpdate(preparedStatement);
+
+			return getHost(newHost.getId(), connection).orElseThrow(()
+					-> new TskCoreException((String.format("Error while fetching newly updated host with id: %d, "))));
+
+		} catch (SQLException ex) {
+			throw new TskCoreException(String.format("Error updating host with name = %s", newHost.getName()), ex);
+		} finally {
+			db.releaseSingleUserCaseWriteLock();
+		}
+	}
+
+	/**
+	 * Delete a host. Name comparison is case-insensitive.
+	 *
+	 * @param name Name of the host to delete.
+	 *
+	 * @return The id of the deleted host or null if no host was deleted.
+	 *
+	 * @throws TskCoreException
+	 */
+	public Long deleteHost(String name) throws TskCoreException {
+		if (name == null) {
+			throw new IllegalArgumentException("Name provided must be non-null");
+		}
+
+		String queryString = "SELECT h.id FROM data_source_info d INNER JOIN tsk_hosts h ON d.host_id = h.id WHERE LOWER(h.name)=LOWER(?)";
+		String deleteString = "DELETE FROM tsk_hosts WHERE LOWER(name) = LOWER(?)";
+
+		CaseDbTransaction trans = this.db.beginTransaction();
+		try {
+			// check if host has any child data sources.  if so, don't delete and throw exception.
```

Comment:
Four other tables have foreign keys referencing tsk_hosts:
tsk_os_account_realms
tsk_os_account_attributes
tsk_os_account_instances
tsk_host_address_map
It's unlikely but possible that these could exist for a host that is no longer associated with a data source (for example, if new rows were created during/right after a merge).

Reply:
Ok.  I'm checking if the host to be deleted is referenced in any of these tables, and if so, I don't delete the host.